core/src/main/java/org/elasticsearch/action/admin/indices/get/TransportGetIndexAction.java

```diff
@@ -69,6 +71,15 @@ protected GetIndexResponse newResponse() {
     }
 
     @Override
+    protected void validateRequest(GetIndexRequest request, ClusterState state) throws Exception {
+        for(String index: request.indices()) {
+            // Create index service also does validation on the index name which does not require
+            // cluster state, so it does not check if things like the index already exists in the cluster
+            MetaDataCreateIndexService.validateIndexOrAliasName(index, InvalidIndexNameException::new);
```

Comment:
I'm not sure where you're headed, but this thing is better done in `org.elasticsearch.action.admin.indices.get.GetIndexRequest#validate`